### PR TITLE
[frontend] Selecting a domain as a nested object of an Ipv4 crashes the front (#6919)

### LINKED ISSUE #6919: Selecting a domain as a nested object of an Ipv4 crashes the front
## Description

When selecting a Domain name Observable to be added as a nested object for an ipv4 address Observable, the front crashes.

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Go to the Knowledge tab of an IP v4 observable
2. click on the + button to add a nested object
3. filter on entity type=domain name
4. select a domain name
5. click on continue

## Expected Output

the drawer to create the nested relationship

## Actual Output

A crash
https://testing.octi.staging.filigran.io/dashboard/observations/observables/1e2e805e-ed88-4642-9a86-077d4f1f367c/knowledge

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

* empty array added for relationshipTypes to avoid crach on R.head(relationshipTypes)
*

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6919
* https://github.com/OpenCTI-Platform/opencti/issues/6959

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Review by marieflorescontact — APPROVED** [review]

File: opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntity.jsx [hunk 1]
```diff
@@ -673,7 +673,8 @@ const StixNestedRefRelationshipCreationFromEntity = ({
   const renderForm = (resolveEntityRef) => {
     let fromEntity = resolveEntityRef.entity;
     let toEntity = targetEntity;
-    let relationshipTypes;
+    let relationshipTypes = [];
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntity.jsx:676]: Not sure to understand the change, let's talk about it.

**Review by Kedae — CHANGES_REQUESTED** [review]

File: opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntity.jsx (hunk 1 above)
Comment (ValentinBouzinFiligran) [reply opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntity.jsx:676]: Just a quick fix
I talked with the product, we will create a new task 

some behaviors are broken

Comment (SouadHadjiat) [reply opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntity.jsx:676]: There was also this open bug https://github.com/OpenCTI-Platform/opencti/issues/6959 for the same issue for which I spent 1h and seeing this same issue in the code, but there are indeed other issues with this screen.

**Comment by ValentinBouzinFiligran** [comment]
This PR is just to avoid the crash
Another one has been created to rework the next part : https://github.com/OpenCTI-Platform/opencti/issues/7064

**Review by Kedae — APPROVED** [review]

**Pull request merged**